dll/win32/shell32/dialogs/drive.cpp

```diff
@@ -234,6 +245,32 @@ SH_ShowDriveProperties(WCHAR *pwszDrive, IDataObject *pDataObj)
     return ret != -1;
 }
 
+BOOL
+SH_ShowDriveProperties(WCHAR *pwszDrive, IDataObject *pDataObj)
+{
+    if (FAILED_UNEXPECTEDLY(SHStrDupW(pwszDrive, &pwszDrive)))
+        return FALSE;
+
+    // Prepare data for thread
+    DRIVE_PROP_DATA *pData = (DRIVE_PROP_DATA *)SHAlloc(sizeof(*pData));
+    if (!pData)
+    {
+        SHFree(pwszDrive);
+        return FALSE;
+    }
+    *pData = { pwszDrive, pDataObj };
```

Comment:
pDataObj is an object you don't control and shell objects are often STA, you can't just pass them to another thread without COM marshaling.

Reply:
@whindsaks I'll leave it up to you to improve this PR. Feel free to copy and modify my code. You understand shell better than I do.

Reply:
No, incorrect code should not be merged in the first place!

Reply:
https://learn.microsoft.com/en-us/windows/win32/shell/ctf
At the end of this page there's a remark about problems of manipulating STA objects in a separate thread.
Some related info:
https://devblogs.microsoft.com/oldnewthing/20151020-00/?p=91321
https://devblogs.microsoft.com/oldnewthing/20151021-00/?p=91311